### PR TITLE
Throw a MissingCredentialsError when credentials are empty/blank

### PR DESCRIPTION
**Pull request by alextwoods**
Fixes: #2281

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

1. To make sure we include your contribution in the release notes, please make sure to add description entry for your changes in the "unreleased changes" section of the `CHANGELOG.md` file (at corresponding gem). For the description entry, please make sure it lives in one line and starts with `Feature` or `Issue` in the correct format.

2. For generated code changes, please checkout below instructions first:
  https://github.com/aws/aws-sdk-ruby/blob/master/CONTRIBUTING.md

Thank you for your contribution!

### CODE REVIEW

File: gems/aws-sigv4/lib/aws-sigv4/signer.rb [hunk 1]
```diff
@@ -654,7 +654,9 @@ def uri_escape_path(string)
 
       def fetch_credentials
         credentials = @credentials_provider.credentials
-        if credentials.access_key_id && credentials.secret_access_key
+        # ensure credentials are not blank
```
Comment (mullermp) [inline gems/aws-sigv4/lib/aws-sigv4/signer.rb:657]: Why not `credentials.set?` If you can't use `.set?` I think regex is a bad idea.. would prefer `nil?` and `empty?`

Comment (alextwoods) [reply gems/aws-sigv4/lib/aws-sigv4/signer.rb:657]: yeah - I was trying to cover the case of blank (eg: '  '), but I agree that's not necessary.  Using `credentials#set?` method now which should work for all credentials/credential providers.

**Comment by alextwoods** [comment]
Ive moved the test out of the S3 client and into core (and added a test both for nil and empty).

**Comment by mullermp** [comment]
Verify that instance profile credentials (and others) still work in these cases.

**Comment by alextwoods** [comment]
The `#set?` method works on all of the credential providers (including InstanceProfileCredentials, ProcessCredentials, ECSCredentials, and the new CognitoIdentityCredentials, ect) - all of these are CredentialProviders that return return `Aws::Credentials`.

**Comment by alextwoods** [comment]
I don't believe we need a changelog for core - I have added specs that test cilent, but I think only the sigv4 gem needs an update.

**Pull request merged**